client/src/components/App/Setup.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import SHA256 from 'crypto-js/sha256';
+import Cookies from 'js-cookie';
+import { register } from '../../actionCreators/auth';
+import Button from '../Button';
+import '../../css/Setup.css';
+
+class Setup extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      privateCode: '',
+      repeatPrivateCode: '',
+      pin: '',
+    };
+  }
+
+  changePrivateCode(e) {
+    this.setState({
+      privateCode: e.target.value,
+    });
+  }
+
+  changeRepeatPrivateCode(e) {
+    this.setState({
+      repeatPrivateCode: e.target.value,
+    });
+  }
+
+  changePin(e) {
+    this.setState({
+      pin: Number(e.target.value),
+    });
+  }
+
+  validate() {
+    const { pin, privateCode, repeatPrivateCode } = this.state;
+    if (privateCode === '') {
+      return '';
+    }
+    if (privateCode.length < 8) {
+      return 'Den personlige koden er for kort (minimum 8 tegn)';
+    }
+    if (privateCode !== repeatPrivateCode) {
+      return 'Personlige koder stemmer ikke overens';
+    }
+    if (pin === '') {
+      return 'Pinkode kan ikke være tom';
+    }
+    return null;
+  }
+
+  submit(e) {
+    e.preventDefault();
+    if (this.validate() === null) {
+      const { pin, privateCode } = this.state;
+      const { username } = this.props;
+      const passwordHash = SHA256(privateCode + username).toString();
+      this.props.register(pin, passwordHash);
+      Cookies.set('passwordHash', passwordHash);
+    }
+  }
+
+  render() {
+    const { privateCode, repeatPrivateCode, pin } = this.state;
+    const errorMessage = this.validate();
+    return (
+      <form className="Setup" onSubmit={(e) => this.submit(e)}>
+        <h2 className="Setup-title">Generalforsamling registrering</h2>
```

Comment:
Norwegian 'særskrivingsfeil'. Either "Generalforsamlingsregistrering" or "Registrering for generalforsamling" or some other title. Yes, I'm being picky.